server/events/vcs/github_client_test.go

```diff
@@ -591,6 +591,157 @@ func TestGithubClient_PullIsMergeable(t *testing.T) {
 	}
 }
 
+func TestGithubClient_PullIsMergeableWithAllowMergeableBypassApply(t *testing.T) {
+	vcsStatusName := "atlantis"
+	cases := []struct {
+		state          string
+		reviewDecision string
+		expMergeable   bool
+	}{
+		{
+			"dirty",
+			`"REVIEW_REQUIRED"`,
+			false,
+		},
+		{
+			"unknown",
+			`"REVIEW_REQUIRED"`,
+			false,
+		},
+		{
+			"blocked",
+			`"REVIEW_REQUIRED"`,
+			false,
+		},
+		{
+			"blocked",
+			`"APPROVED"`,
+			true,
+		},
+		{
+			"blocked",
+			"null",
+			true,
+		},
+		{
+			"behind",
+			`"REVIEW_REQUIRED"`,
+			false,
+		},
+		{
+			"random",
+			`"REVIEW_REQUIRED"`,
+			false,
+		},
+		{
+			"unstable",
+			`"REVIEW_REQUIRED"`,
+			true,
+		},
+		{
+			"has_hooks",
+			`"APPROVED"`,
+			true,
+		},
+		{
+			"clean",
+			`"APPROVED"`,
+			true,
+		},
+		{
+			"",
+			`"APPROVED"`,
+			false,
+		},
+	}
+
+	// Use a real GitHub json response and edit the mergeable_state field.
+	jsBytes, err := os.ReadFile("fixtures/github-pull-request.json")
+	Ok(t, err)
+	prJSON := string(jsBytes)
+
+	// Status Check Response
+	jsBytes, err = os.ReadFile("fixtures/github-commit-status-full.json")
+	Ok(t, err)
+	commitJSON := string(jsBytes)
+
+	// Branch protection Response
+	jsBytes, err = os.ReadFile("fixtures/github-branch-protection-required-checks.json")
+	Ok(t, err)
+	branchProtectionJSON := string(jsBytes)
+
+	// Branch protection Response
```

Comment:
Just nit. I'm guessing you meant something like `List check suites Response` in this comment?

Reply:
yeah, copy paste :(

Reply:
Do you want to fix it before we include it in the next pre-release?

Reply:
Yeah, right., will fix now

Reply:
It's updated now

Reply:
Nice thanks a lot!